common/intents/intent.go

```diff
@@ -41,6 +41,13 @@ type FileNeedsIOBuffer interface {
 	ReleaseIOBuffer()
 }
 
+// IndexDocument holds information about a collection's index.
+type IndexDocument struct {
```

Comment:
it feels a little odd to have this in a package called `intents`, especially when it's imported from elsewhere. Perhaps we could put this into `common/db` or into a new package like `common/metadata` or `common/indexes`?

Reply:
Yeah good point. I moved it to `common/db`.